src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -369,6 +370,15 @@ private ConstantValue GetLazyConstantValue(bool earlyDecodingWellKnownAttributes
             Debug.Assert((GetLazyConstantValue(earlyDecodingWellKnownAttributes) == Microsoft.CodeAnalysis.ConstantValue.Unset) ||
                 (GetLazyConstantValue(earlyDecodingWellKnownAttributes) == value));
 
+            // Check availability of `DecimalConstantAttribute`.
+            if (value is { IsDecimal: true })
+            {
+                Binder.ReportUseSiteDiagnosticForSynthesizedAttribute(DeclaringCompilation,
```

Comment:
>Binder.ReportUseSiteDiagnosticForSynthesizedAttribute(DeclaringCompilation [](http://example.com/codeflow?start=16&length=74)

It looks like the purpose of the enclosing method is to store result of the work performed elsewhere rather than performing any extra checks and adding to the diagnostic bag. I think the caller should be responsible for doing this.